### PR TITLE
[Bug] Add auto increment primary key to notes_data table

### LINKED ISSUE #15064: [Bug]: Deadlock on notes_data
### Pimcore version

10.x or 11.x

### Steps to reproduce

On high concurrency systems where lot of notes are written, the table ends up locking all the time. Reason is cause it uses id and name is primary index. If we change the index to be auto-incremented, it doesn't deadlock anymore.

Current Table is:

```
CREATE TABLE `notes_data` (
  `id` int(11) NOT NULL,
  `name` varchar(255) NOT NULL,
  `type` enum('text','date','document','asset','object','bool') DEFAULT NULL,
  `data` text,
  PRIMARY KEY (`id`, `name`)
) DEFAULT CHARSET=utf8mb4;
```

### Expected / actual behavior

Better Layout would be:

```
CREATE TABLE `notes_data` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `cid` int(11) NOT NULL,
  `name` varchar(255) NOT NULL,
  `type` enum('text','date','document','asset','object','bool') DEFAULT NULL,
  `data` text DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `id_cid` (`id`,`cid`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;

```

### PR DESCRIPTION
**Pull request by dpfaffenbauer**

### CODE REVIEW

**Comment by fashxp** [comment]
Do we need to adapt install.sql too?

**Comment by dpfaffenbauer** [comment]
Obviously yes, I forgot about it. Will add it too

**Comment by dpfaffenbauer** [comment]
@fashxp added it

File: bundles/CoreBundle/Migrations/Version20230428112302.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        $notesData = $schema->getTable('notes_data');
+
+        if (!$notesData->hasColumn('auto_id')) {
+            $notesData->addColumn('auto_id', 'integer', [
+                'autoincrement' => true,
+            ]);
+
+            $notesData->dropPrimaryKey();
+            $notesData->setPrimaryKey(['auto_id']);
+            $notesData->addUniqueIndex(['id', 'name']);
```
Comment (mcop1) [inline bundles/CoreBundle/Migrations/Version20230428112302.php]:
```suggestion
            $notesData->addUniqueIndex(['id', 'name'], 'UNIQ_E5A8E5E2BF3967505E237E06');
```

File: bundles/CoreBundle/Migrations/Version20230428112302.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public function down(Schema $schema): void
+    {
+        $notesData = $schema->getTable('notes_data');
+
+        if ($notesData->hasColumn('auto_id')) {
+            $notesData->dropPrimaryKey();
+            $notesData->dropColumn('auto_id');
+            $notesData->setPrimaryKey(['id', 'name']);
```
Comment (mcop1) [inline bundles/CoreBundle/Migrations/Version20230428112302.php:39]:
```suggestion
            $notesData->setPrimaryKey(['id', 'name']);
            $notesData->dropIndex('UNIQ_E5A8E5E2BF3967505E237E06');
```

**Review by mcop1 — COMMENTED** [review]
@dpfaffenbauer Should we add a name to the index? That would allow us to drop the index in the down migration. What do you think?

**Comment by dpfaffenbauer** [comment]
@mcop1 done

**Comment by mcop1** [comment]
Thanks!

**Pull request merged**